molter/help.py

```diff
@@ -0,0 +1,187 @@
+
+    embed_title: str = attrs.field(default="{username} Help Command", kw_only=True)
+    """The title to use in the embed. {username} will be replaced by the client's username"""
+    not_found_message: str = attrs.field(default="Sorry! No command called `{cmd_name}` was found.", kw_only=True)
+    """The message to send when a command was not found. {cmd_name} will be replaced by the requested command."""
+
+    _client: "Snake" = attrs.field()
+    _cmd: molter.MolterCommand | None = attrs.field(init=False, default=None)
+
+    def __attrs_post_init__(self) -> None:
+        if not self._cmd:
+            self._cmd = self._callback  # type: ignore
+
+    def register(self) -> None:
+        """Register the help command in dis-snek"""
+        if not isinstance(self._cmd.callback, functools.partial):
+            # prevent wrap-nesting
+            self._cmd.callback = functools.partial(self._cmd.callback, self)
+
+        # replace existing help command if found
+        if "help" in self._client.commands:
+            log.warning("Replacing existing help command.")
+            del self._client.commands["help"]
+
+        self._client.add_message_command(self._cmd)  # type: ignore
+
+    async def send_help(self, ctx: MessageContext, cmd_name: str | None) -> None:
+        """
+        Send a help message to the given context.
+
+        args:
+            ctx: The context to use.
+            cmd_name: An optional command name to send help for.
+        """
+        await self._callback.callback(ctx, cmd_name)  # type: ignore
+
+    @molter.msg_command(name="help")
+    async def _callback(self, ctx: MessageContext, *, cmd_name: str = None) -> None:
+        if cmd_name:
+            return await self._help_specific(ctx, cmd_name)
+        await self._help_list(ctx)
+
+    async def _help_list(self, ctx: MessageContext) -> None:
+        cmds = await self._gather(ctx)
+
+        output = []
+        for cmd in cmds.values():
+            _temp = self._generate_command_string(cmd, ctx)
+            _temp += f"\n{cmd.brief}"
+
+            output.append(self._sanitise_mentions(_temp))
+        if len("\n".join(output)) > 500:
+            paginator = Paginator.create_from_list(self._client, output, page_size=500)
+            paginator.default_color = self.embed_color
+            paginator.default_title = self.embed_title.format(username=self._client.user.username)
+            await paginator.send(ctx)
+        else:
+            embed = Embed(
+                title=self.embed_title.format(username=self._client.user.username),
+                description="\n".join(output),
+                color=self.embed_color,
+            )
+            await ctx.reply(embeds=embed)
+
+    async def _help_specific(self, ctx: MessageContext, cmd_name: str) -> None:
+        cmds = await self._gather(ctx)
+
+        if cmd := cmds.get(cmd_name.lower()):
+            _temp = self._generate_command_string(cmd, ctx)
+            _temp += f"\n{cmd.help}"
```

Comment:
Ditto for `cmd.help`. If you want to add a user-provided placeholder, this and the `cmd.brief` one can easily be the same string.